### PR TITLE
[Azure] update azure catalog and add westus3

### PR DESCRIPTION
**Pull request by Michaelvll**
Related to https://github.com/skypilot-org/skypilot/pull/2149

Future TODO:
- [ ] Auto update of the Azure catalog

### CODE REVIEW

File: catalogs/v5/azure/vms.csv [hunk 1]
```diff
@@ -1,164 +1,171 @@
...
+Standard_A4,,,8.0,14,,0.405,0.115546,westus2,V1
+Standard_A4m_v2,,,4.0,32,,0.27,0.027756,centralus,V1
+Standard_A4m_v2,,,4.0,32,,0.238,0.070853,eastus,V1
+Standard_A4m_v2,,,4.0,32,,0.249,0.087922,eastus2,V1
+Standard_A4m_v2,,,4.0,32,,0.27,0.027891,northcentralus,V1
+Standard_A4m_v2,,,4.0,32,,0.248,0.02723,southcentralus,V1
+Standard_A4m_v2,,,4.0,32,,0.249,0.025821,westcentralus,V1
+Standard_A4m_v2,,,4.0,32,,0.297,0.030353,westus,V1
+Standard_A4m_v2,,,4.0,32,,0.208,0.02211,westus2,V1
+Standard_A4m_v2,,,4.0,32,,,,WestUS3,V1
```
Comment (WoosukKwon) [inline catalogs/v5/azure/vms.csv:117]: It seems the westus3 VMs do not have price information. Is this correct?

**Review by WoosukKwon — APPROVED** [review]
Thanks for updating it. Just a heads up: It seems the westus3 VMs do not have price information. Is this correct?

File: catalogs/v5/azure/vms.csv (hunk 1 above)
Comment (Michaelvll) [reply catalogs/v5/azure/vms.csv:117]: Ahh, good catch! Let me double check.

Comment (Michaelvll) [reply catalogs/v5/azure/vms.csv]: That was caused by the case of the region name. Just fixed. PTAL. : )

**Review by WoosukKwon — APPROVED** [review]
LGTM!

**Pull request merged**